### PR TITLE
Ddfbra 201 go anonym og indlogget bruger kan prove lydbog

### PR DESCRIPTION
**Pull request by ThomasGross**
#### Link to issue

https://reload.atlassian.net/browse/DDFBRA-201

#### Description

This PR includes 
- A publizon player module. 
- Shared dialog and drawer component

### CODE REVIEW

File: __tests__/url.test.ts [hunk 1]
```diff
@@ -3,20 +3,19 @@ import { expect, test } from "vitest"
...
-    routeParams: { wid: 123 },
+    routeParams: { work: "work", wid: 123 },
     queryParams: { audio: "true" },
   })
   expect(workUrl).toBe("/work/123?audio=true")
 })
 
 test("That resolveUrl can return a search url", async () => {
-  const workUrl = resolveUrl({ type: "search", queryParams: { q: "test" } })
+  const workUrl = resolveUrl({ routeParams: { search: "search" }, queryParams: { q: "test" } })
```
Comment (spaceo) [inline __tests__/url.test.ts:19]: We talked about taking a look at the `resolveUrl` :) Let's do that on a rainy day.

File: components/pages/workPageLayout/WorkPageLayout.tsx [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+
+  const { data } = useQuery({
     queryKey: useGetMaterialQuery.getKey({ wid }),
     queryFn: useGetMaterialQuery.fetcher({ wid }),
   })
 
+  const manifestations = data?.work?.manifestations.all
+
+  const identifier = manifestations?.[0].identifiers?.[0].value || ""
+
```
Comment (spaceo) [inline components/pages/workPageLayout/WorkPageLayout.tsx:24]: What happens if we do not get data or work and therefore do not have manifestations? How is the rest of the code reacting?

File: components/shared/publizonPlayer/helper.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    document.head.appendChild(scriptElement)
+  }
+}
+
+export const removeAsset = ({ src, type }: TAssetType) => {
+  if (type === "script") {
+    const scriptElement = document.querySelector(`script[src="${src}"]`)
+    scriptElement?.remove()
+  }
+}
```
Comment (spaceo) [inline components/shared/publizonPlayer/helper.ts:29]: Why having two sets of `appendAsset` `removeAsset` functions?
As far as I see we can a common way of doing this in `/lib/helpers/reader-player/...` and share it among the components.
You could consider making your custom hook adding the assets receiving an array of assets - which you could use in both the reader and player?

**Review by spaceo — APPROVED** [review]
I think you could consider consolidating functionality that is repeated in both the reader and player and define it once.
Othjerwise, looks good 👍

File: components/pages/workPageLayout/WorkPageLayout.tsx (hunk 2 above)
Comment (ThomasGross) [reply components/pages/workPageLayout/WorkPageLayout.tsx]: TODO is added here

File: components/shared/publizonPlayer/helper.ts (hunk 3 above)
Comment (ThomasGross) [reply components/shared/publizonPlayer/helper.ts]: Great idea. I will do that and merge the branch into main afterwards.

**Pull request merged**